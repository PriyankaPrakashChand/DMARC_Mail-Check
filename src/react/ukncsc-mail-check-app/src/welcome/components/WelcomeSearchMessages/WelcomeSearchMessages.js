import React from 'react';
import { Divider } from 'semantic-ui-react';
import { MailCheckMessage } from 'common/components';

export default ({
  addDomainOnClick,
  error,
  hasSearched,
  lastSearchTerm,
  loading,
  searchTermIsPublicSectorOrg,
}) => {
  if (loading || !hasSearched) {
    return null;
  }

  if (error) {
    return (
      <React.Fragment>
        <MailCheckMessage error>{error.message}</MailCheckMessage>
        <Divider hidden />
      </React.Fragment>
    );
  }

  if (searchTermIsPublicSectorOrg) {
    return (
      <React.Fragment>
        <MailCheckMessage info>
          {lastSearchTerm} is not currently monitored by Mail Check.{' '}
          <a href="" onClick={addDomainOnClick}>
            Add {lastSearchTerm} to Mail Check
          </a>.
        </MailCheckMessage>
        <Divider hidden />
      </React.Fragment>
    );
  }

  return (
    <React.Fragment>
      <MailCheckMessage info>
        {lastSearchTerm} is not currently monitored by Mail Check. Please email{' '}
        <a
          href={`mailto:mailcheck@digital.ncsc.gov.uk?Subject=Add ${lastSearchTerm} to Mail Check`}
        >
          mailcheck@digital.ncsc.gov.uk
        </a>{' '}
        to have {lastSearchTerm} added to Mail Check.
      </MailCheckMessage>
      <Divider hidden />
    </React.Fragment>
  );
};
