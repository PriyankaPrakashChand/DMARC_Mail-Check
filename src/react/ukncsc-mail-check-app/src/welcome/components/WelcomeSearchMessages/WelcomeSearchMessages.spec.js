import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';
import WelcomeSearchMessages from './WelcomeSearchMessages';

describe('WelcomeSearchMessages', () => {
  let container;

  describe('when loading', () => {
    beforeEach(() => {
      ({ container } = render(<WelcomeSearchMessages loading />));
    });

    test('it should render nothing', () =>
      expect(container.firstChild).toBeNull());
  });

  describe('when no search has happened', () => {
    beforeEach(() => {
      ({ container } = render(<WelcomeSearchMessages hasSearched={false} />));
    });

    test('it should render nothing', () =>
      expect(container.firstChild).toBeNull());
  });

  describe('when an error occurs', () => {
    beforeEach(() => {
      ({ container } = render(
        <WelcomeSearchMessages
          error={{ message: 'something has gone wrong' }}
          hasSearched
          loading={false}
        />
      ));
    });

    test('it should render the error message', () =>
      expect(container.firstChild).toHaveTextContent(
        'something has gone wrong'
      ));

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when there is no result for a public sector domain', () => {
    beforeEach(() => {
      ({ container } = render(
        <WelcomeSearchMessages
          hasSearched
          lastSearchTerm="foo.gov.uk"
          loading={false}
          searchResult={null}
          searchTermIsPublicSectorOrg
        />
      ));
    });

    test('it should render the please add to mail check message', () =>
      expect(container.firstChild).toHaveTextContent(
        'foo.gov.uk is not currently monitored by Mail Check. Add foo.gov.uk to Mail Check.'
      ));

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when is no result for a non public sector domain', () => {
    beforeEach(() => {
      ({ container } = render(
        <WelcomeSearchMessages
          hasSearched
          loading={false}
          lastSearchTerm="foo.com"
          searchResult={null}
        />
      ));
    });

    test('it should render please contact mail check message', () =>
      expect(container.firstChild).toHaveTextContent(
        'foo.com is not currently monitored by Mail Check. Please email mailcheck@digital.ncsc.gov.uk to have foo.com added to Mail Check.'
      ));

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
