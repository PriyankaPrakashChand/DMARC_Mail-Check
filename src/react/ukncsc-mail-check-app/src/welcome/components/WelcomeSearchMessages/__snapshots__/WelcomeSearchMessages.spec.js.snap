// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSearchMessages when an error occurs it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui error message MailCheckMessage--container"
  >
    something has gone wrong
  </div>
  <div
    class="ui hidden divider"
  />
</div>
`;

exports[`WelcomeSearchMessages when is no result for a non public sector domain it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui info message MailCheckMessage--container"
  >
    foo.com
    is not currently monitored by Mail Check. Please email
    <a
      href="mailto:mailcheck@digital.ncsc.gov.uk?Subject=Add foo.com to Mail Check"
    >
      mailcheck@digital.ncsc.gov.uk
    </a>
    to have 
    foo.com
    added to Mail Check.
  </div>
  <div
    class="ui hidden divider"
  />
</div>
`;

exports[`WelcomeSearchMessages when there is no result for a public sector domain it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui info message MailCheckMessage--container"
  >
    foo.gov.uk
    is not currently monitored by Mail Check.
    <a
      href=""
    >
      Add 
      foo.gov.uk
      to Mail Check
    </a>
    .
  </div>
  <div
    class="ui hidden divider"
  />
</div>
`;
